Replace "btn-border" with USWDS checkbox and radio

aduth opened the pull request:
**Why**:

- Consistency and alignment toward design system
- Semantic correctness in considering checkboxes and radio inputs as inputs, not as buttons
- Eliminates now-unused custom styling and JavaScript
- Avoids relying on JavaScript for stylistic effects
- Simpler implementation in some cases

**Implementation Notes:**

As of `identity-style-guide@6` (incorporated in #5265), we now have access to a set of [stylized radio and checkbox inputs](https://design.login.gov/components/form-fields/#radio-buttons), including bordered and tile variations, which can serve in place of the custom `btn-border` class.

**Screenshots:**

Screen|Before|After
---|---|---
Sign in|![Screen Shot 2021-08-10 at 11 04 30 AM](https://user-images.githubusercontent.com/1779930/128892666-413ed6ef-bf37-4ef8-b629-980623f9563c.png)|![Screen Shot 2021-09-08 at 10 47 00 AM](https://user-images.githubusercontent.com/1779930/132532362-4a69a4ae-a766-48cb-8652-75fa0da70832.png)
OTP Selection|![Screen Shot 2021-08-10 at 11 04 18 AM](https://user-images.githubusercontent.com/1779930/128892741-ee50e773-ade8-4662-8861-2bf74ff8ced3.png)|![Screen Shot 2021-09-08 at 10 55 51 AM](https://user-images.githubusercontent.com/1779930/132533463-4a7f0cf7-581a-4819-b972-d19aa4647d13.png)
OTP Setup|![Screen Shot 2021-08-10 at 11 05 07 AM](https://user-images.githubusercontent.com/1779930/128892796-c680cd5d-4985-41fe-a067-eca9499d57ec.png)|![Screen Shot 2021-09-08 at 10 44 09 AM](https://user-images.githubusercontent.com/1779930/132532618-da8e2d0b-6289-4d71-8a71-86929f59d74a.png)
Add phone|![Screen Shot 2021-08-10 at 11 14 54 AM](https://user-images.githubusercontent.com/1779930/128893003-b14b611c-781e-4d93-b556-9bf42ac989ae.png)|![Screen Shot 2021-09-08 at 10 45 02 AM](https://user-images.githubusercontent.com/1779930/132532556-5550bd45-ccbe-453e-9989-54091f1a52dc.png)
IAL2 Agreement|![Screen Shot 2021-08-10 at 11 05 34 AM](https://user-images.githubusercontent.com/1779930/128893053-cf5a9b28-620f-4bb0-8c62-b73b71c8a79b.png)|![Screen Shot 2021-09-08 at 10 45 54 AM](https://user-images.githubusercontent.com/1779930/132532530-482378a9-3cc3-48f2-a7c2-9184248999ea.png)
IAL2 SSN Toggle|![Screen Shot 2021-08-10 at 11 05 51 AM](https://user-images.githubusercontent.com/1779930/128893113-03ad2023-18bb-47ca-ac7b-0b11eca34c31.png)|![Screen Shot 2021-09-08 at 10 46 16 AM](https://user-images.githubusercontent.com/1779930/132532493-251b20f9-2457-4254-bbf4-f64f13d3c3a4.png)

zachmargolis commented on app/views/idv/doc_auth/agreement.html.erb:
```diff
@@ -19,14 +19,17 @@
+    class="usa-checkbox__label margin-bottom-4"
+    for="ial2_consent_given"
+  ><%= t('doc_auth.instructions.consent') %>
+    <%= new_window_link_to t('doc_auth.instructions.learn_more'), 'https://login.gov/policy/' %>
```
I know this URL was already like this but can we use our method for this?

```suggestion
    <%= new_window_link_to t('doc_auth.instructions.learn_more'), MarketingSite.security_and_privacy_practices_url %>
```

aduth replied:
> I know this URL was already like this but can we use our method for this?

Good call, updated in d58f2e18f.

zachmargolis commented on app/views/idv/otp_delivery_method/new.html.erb:
```diff
@@ -24,29 +24,28 @@
+          false,
+          class: 'otp_delivery_preference_voice usa-radio__input usa-radio__input--tile'
+        ) %>
+    <label class="usa-radio__label"><%= t('two_factor_authentication.otp_delivery_preference.voice') %>
```
WDYT of adding a newline?
```suggestion
    <label class="usa-radio__label">
      <%= t('two_factor_authentication.otp_delivery_preference.voice') %>
```

aduth replied:
This is one of the rare, unfortunate cases where whitespace matters in HTML. The newline collapses down to a single space, which throws off the layout.

I'd definitely agree it would hopefully be something we shouldn't need to worry about. I'm thinking about a few ways we could address this:

- Try to control whitespace behavior with CSS styling
- Use utilities like `label_tag`, etc. and avoid writing out the markup

zachmargolis replied:
I think CSS styling would be ideal, or maybe we go with a comment approach

```suggestion
    <label class="usa-radio__label"><!-- can't have leading whitespace
      --><%= t('two_factor_authentication.otp_delivery_preference.voice') %>
```

aduth replied:
Follow-up at https://github.com/uswds/uswds/pull/4286. Still on the fence for how to handle it in the short-term.

anniehirshman-gsa replied:
> @anniehirshman-gsa Good question. I would be fine with updating this in the design system. Might be worth a check with Chanan if it'd have any impact on her work on the upcoming partners site.

@aduth confirming that the larger text size (1.125 rem) is approved on the UX side for tiled radio buttons and checkboxes, and I've updated Figma. Figma also reflects the larger padding that would be great to update as well.

aduth replied:
Thanks @anniehirshman-gsa . I've got a [small conundrum to sort out with the whitespace improvements](https://github.com/uswds/uswds/pull/4286#discussion_r686981332) from this thread, but otherwise my plan will be to incorporate those changes as a temporary patch into the design system. I can make the tile font size changes around the same time. Since all this is non-ticketed refactoring, I might be a bit delayed in getting to it for prioritization's sake.

zachmargolis replied:
If we want to merge this sooner rather than later, I think it might be nice to leave the commented version of the newline, and then we could leave in a "fix after USDWS update" in there too so we can find them easily if/when USWDS is handled to be more tolerant of whitespace

zachmargolis commented on app/views/sign_up/registrations/new.html.erb:
```diff
@@ -47,8 +47,7 @@
                                          required: true, aria: { invalid: false } }, true, false %>
-      <label for="user_terms_accepted" class="usa-checkbox__label">
-        <%= t('sign_up.terms') %>
+      <label for="user_terms_accepted" class="usa-checkbox__label"><%= t('sign_up.terms') %>
```
did one of the linters move the text up?

zachmargolis commented on app/views/users/shared/_otp_make_default_number.html.erb:
```diff
@@ -11,13 +12,14 @@
+          form_obj.otp_make_default_number,
+          class: 'usa-checkbox__input usa-checkbox__input--bordered',
+        ) %>
+    <label class="usa-checkbox__label" for="<%= form_name_tag %>">
```
is the `for` here needed for CSS or is it just for accessibility best practices?

aduth replied:
> is the `for` here needed for CSS or is it just for accessibility best practices?

It's needed both for accessibility and also to maintain the behavior that clicking the label should toggle the checkbox.

Previously we were relying on [implicit label association](https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly) by using the label to wrap both the input and label. That's not an option for the design system checkboxes.

zachmargolis commented on .erb-lint.yml:
```diff
@@ -60,7 +60,6 @@ linters:
           - '((sm|md|lg)-)?hide'
           - '(sm|md|lg)-show'
           - 'btn(-(small|big|narrow|wide|link|primary|secondary|danger|disabled|big|narrow|transparent))?'
-          - ':not(label).btn-border'
```
is this removed because we don't use the `btn-border` style any more? or because the style no longer exists?

aduth replied:
> is this removed because we don't use the `btn-border` style any more? or because the style no longer exists?

I could have kept it and removed the one exception we'd previously allowed:

```yml
- '.btn-border'
```

This would also be most consistent with how we've used the deprecated class linting to date. I think I've just become less convinced over time of the value of these deprecations. Because the class no longer exists, it'll be obviously visually broken if someone tries to use it.  It really only prevents a case where someone is copy-pasting code, which itself is unlikely because there will no longer be any remnants of the class in the codebase after its removal.

That being said, I'll probably re-add it, just for consistency's sake, and we can decide separately if / how to deal with deprecations.

aduth replied:
> That being said, I'll probably re-add it, just for consistency's sake, and we can decide separately if / how to deal with deprecations.

Re-added in e519dc7ec.

zachmargolis reviewed (approved):
LGTM

anniehirshman-gsa reviewed (commented):
Screenshots and checkboxes are looking good!

For the tiled radio groups: the IdP uses a slightly larger size for the blue text/label that I'd like to retain for emphasis, even though I know it's not in the design system right now.  (I used 1.125 rem aka h3 size in the Public Sans update.) Any suggestions on the best way to implement that?  Also looping in @nickttng for opinions if you have

aduth commented:
> For the tiled radio groups: the IdP uses a slightly larger size for the blue text/label that I'd like to retain for emphasis, even though I know it's not in the design system right now. (I used 1.125 rem aka h3 size in the Public Sans update.) Any suggestions on the best way to implement that?

Should it be IdP-specific, or would it make sense to update it in the design system?

anniehirshman-gsa commented:
> > For the tiled radio groups: the IdP uses a slightly larger size for the blue text/label that I'd like to retain for emphasis, even though I know it's not in the design system right now. (I used 1.125 rem aka h3 size in the Public Sans update.) Any suggestions on the best way to implement that?
> 
> Should it be IdP-specific, or would it make sense to update it in the design system?

All of the examples I know of in the IdP have the larger text size, so I would be fine with updating this in the design system.  @nickttng do you think we should go ahead and make this change in the design system, keep it IdP-specific, throw it out to the wider group for discussion?

aduth commented:
I rebased the branch to resolve conflicts and updated screenshots in the original comment to reflect pending changes from upcoming `identity-style-guide@6.1.0`.

A couple things to complete before this is ready:

- [ ] Publish `identity-style-guide@6.1.0`
- [ ] In separate pull request, update `identity-idp` to use `identity-style-guide@6.1.0`

anniehirshman-gsa reviewed (approved):
LGTM!
